nav2_regulated_pure_pursuit_controller/src/regulated_pure_pursuit_controller.cpp

```diff
@@ -508,8 +508,10 @@ bool RegulatedPurePursuitController::isCollisionImminentExtendedSearch()
     std::back_inserter(extended_collision_check_path.poses),
     [](geometry_msgs::msg::PoseStamped & poseStamped) {
       // Modify Z For visualization in rviz so it is drawn between global plan and collision arc
+      geometry_msgs::msg::PoseStamped transformed_pose;
       poseStamped.pose.position.z = 0.005;
-      return poseStamped;
+      transformPose(costmap_ros_->getBaseFrameID(), poseStamped, transformed_pose);
```

Comment:
Why do we transform it to the base frame, not to the global (which is `odom` after bringup is merged)

Reply:
> Good point.. I wonder if collision checking that happens in the later lines works either way

I checked and was looking not bad but I´d like to know more.